tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeSpacetimeQuantities.cpp

```diff
@@ -380,6 +385,50 @@ void test_compute_3d_extrinsic_curvature(const DataVector& used_for_size) {
       extrinsic_curvature);
 }
 
+template <size_t Dim, typename T>
+void test_compute_spatial_metric_lapse_shift(const T& used_for_size) {
+  // Set up random values for lapse, shift, and spatial_metric.
+  std::random_device r;
+  const auto seed = r();
+  std::mt19937 generator(seed);
+  INFO("seed" << seed);
+  std::uniform_real_distribution<> dist(-1., 1.);
+  std::uniform_real_distribution<> dist_positive(1., 2.);
+  const auto nn_generator = make_not_null(&generator);
+  const auto nn_dist = make_not_null(&dist);
+  const auto nn_dist_positive = make_not_null(&dist_positive);
+
+  const auto lapse = make_with_random_values<Scalar<T>>(
+      nn_generator, nn_dist_positive, used_for_size);
+  const auto shift = make_with_random_values<tnsr::I<T, Dim>>(
+      nn_generator, nn_dist, used_for_size);
+  const auto spatial_metric = [&]() {
+    auto spatial_metric_l = make_with_random_values<tnsr::ii<T, Dim>>(
+        nn_generator, nn_dist, used_for_size);
+    // Make sure spatial_metric isn't singular by adding
+    // large enough positive diagonal values.
+    for (size_t i = 0; i < Dim; ++i) {
+      spatial_metric_l.get(i, i) += 4.0;
+    }
+    return spatial_metric_l;
+  }();
+
+  // Make spacetime metric from spatial metric, lapse, and shift.
+  // Then go backwards and compute the spatial metric, lapse, and shift
+  // and make sure we get back the original values.
+  const auto psi = gr::spacetime_metric(lapse, shift, spatial_metric);
+
+  // Here are the functions we are testing.
+  const auto spatial_metric_test = gr::spatial_metric(psi);
+  const auto shift_test =
+      gr::shift(psi, determinant_and_inverse(spatial_metric).second);
+  const auto lapse_test = gr::lapse(shift, psi);
+
```

Comment:
If you want to make changes, couldn't you just put these function calls into the CHECK_ITERABLE_APPROX instead of storing to variables? I don't think it's a big deal, though.

Reply:
Yes I could.  Not sure what is preferred.  I find the code more readable with these variables rather than with a huge messy function call, but if others disagree I am happy to change it.